build/gitea-fmt.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -e
+
+GO="${GO:=go}"
+
+# a simple self-check, make sure the current working directory is Gitea's repo
+if [ ! -f ./build/gitea-fmt.sh ]; then
+  echo "$0 could only run in Gitea's source directory"
+  exit 1
+fi
+
+if [ "$1" != "-l" -a "$1" != "-w" ]; then
+  echo "$0 could only accept '-l' (list only) or '-w' (write to files) argument"
+  exit 1
+fi
+
+GO_VERSION=$(grep -Eo '^go\s+[0-9]+\.[0-9]+' go.mod | cut -d' ' -f2)
+
+echo "Run gofumpt with Go language version $GO_VERSION ..."
+gofumpt -extra -lang "$GO_VERSION" "$1" .
```

Comment:
It should use the version specified in Makefiles, so you would still need to pass `GOFUMPT_PACKAGE` to this script.

Reply:
I would like to assume the the gofump has been installed by line 810 `$(GO) install $(GOFUMPT_PACKAGE)`

Otherwise the line 810 should be removed.

Feel free to propose changes to this PR.

Reply:
It could still run a different version based on what the user had installed last. I think we need to remove these `go install` lines and use `go run` only to ensure we get the correct versions, but I can do that in another PR.

Reply:
After some more thinking, I think I would prefer to use pre-installed gofumpt.

The reason is: Gitea had better to manage all its dependent binaries with correct version and have them in local cache. Otherwise, if developer's network is not stable or they are offline, the `go run` will be time wasting for their `make fmt` work.

Reply:
I mean something like you said, "manage all its dependent binaries with correct version and have them in local cache", like `./.../bin/gofumpt-0.x.x`